docs/introduction/index.md

```diff
@@ -1,26 +1,255 @@
+the [Model Definition](/docs/guides/model-definition.html) guide.
+
+### Adding business logic
+
+The business logic of a Bounded Context is based on [Entities](#entities).
+They handle messages updating the state in response. Entities like `Aggregate`s and
+`ProcessManager`s can generate events. `ProcessManager`s can also generate new commands.
+`Projection`s only consume events. 
+
+Updating the state of the domain model in response to messages and generating new messages is
+the “life” of the domain model. Messages are delivered to entities by [Repositories](#repositories).
+
+#### Entities
+
+During this step we create entity classes and add message handling methods to them. 
+Code snippets below show `Aggregate` and `Projection` classes with their handler methods.
+
+```java
+final class TaskAggregate
+    extends Aggregate<TaskId, Task, Task.Builder> {
+    
+    @Assign
+    TaskCreated handle(CreateTask cmd, CommandContext ctx) {
+        return TaskCreated
+                .newBuilder()
+                .setId(cmd.getId())
+                .setName(cmd.getName())
+                .setDescription(cmd.getDescription())
+                .setWhoCreated(ctx.getActor())
+                .vBuild();
+    }
+    
+    @Apply
+    void event(TaskCreated e) {
+        builder().setName(e.getName())
+                 .setDescription(e.getDescription())
+                 .setOwner(e.getWhoCreated());
+    }
+}
+```
 
-We started working on Spine after two CQRS/ES projects and having the experience
-of tons of manual work for:
- - creating commands, events, etc.
- - delivering events and data to web and mobile clients.
+```java
+final class TaskItemProjection
+    extends Projection<TaskId, TaskItem, TaskItem.Builder> {
 
-We decided to create a framework that can help us and development groups like us
-building apps using Domain-Driven Design. These are the principles our framework is based on:
+    @Subscribe
+    void on(TaskCreated e) {
+        builder().setId(e.getId())
+                 .setName(e.getName())
+    }
 
- - Events and Commands should be **strongly typed**.
- - It should be **easy to pass data** between different programming languages and platforms. 
-   We want a domain language to become truly Ubiquitous.
- - Objects must be **immutable** unless there is a good reason not to do so.
- - Developers should spend more time on business logic rather than plumbing related to storage
-   matters or server environment aspects.
- - The code should be easier to test.
+    @Subscribe
+    void on(TaskCompleted e, EventContext ctx) {
+        builder().setWhenDone(ctx.getTimestamp());
+    }
+}
+```
+
+#### Repositories
+The framework provides default implementations for repositories.
+A custom `Repository` class may be needed for:
+  * <strong>Dispatching messages to entities in a non-standard way</strong>.
+    By default, a command is dispatched using the first field of the command message.
```

Comment:
`...using the first field of the command message as an ID of the target entity.`